packages/workspace/src/browser/workspace-service.ts

```diff
@@ -120,22 +123,29 @@ export class WorkspaceService implements FrontendApplicationContribution {
     }
 
     protected async updateRoots(): Promise<void> {
-        this._roots = await this.computeRoots();
+        const computedRoots = await this.computeRoots();
+        this._roots = computedRoots || [];
+        if (!computedRoots) {
+            this._workspace = undefined;
```

Comment:
should we move this into the init() function, to let updateRoots() only deal with roots?